packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -326,32 +339,48 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     ? localScreenShareStreamComponent
     : undefined;
 
-  const videoGalleryLayout =
-    layout === 'floatingLocalVideo' ? (
-      <FloatingLocalVideoLayout
-        remoteParticipants={remoteParticipants}
-        onRenderRemoteParticipant={onRenderRemoteVideoTile ?? defaultOnRenderVideoTile}
-        localVideoComponent={localVideoTile}
-        screenShareComponent={screenShareComponent}
-        showCameraSwitcherInLocalPreview={showCameraSwitcherInLocalPreview}
-        maxRemoteVideoStreams={maxRemoteVideoStreams}
-        dominantSpeakers={dominantSpeakers}
-        parentWidth={containerWidth}
-        parentHeight={containerHeight}
-        styles={styles}
-      />
-    ) : (
-      <DefaultLayout
-        remoteParticipants={remoteParticipants}
-        onRenderRemoteParticipant={onRenderRemoteVideoTile ?? defaultOnRenderVideoTile}
-        localVideoComponent={localVideoTile}
-        screenShareComponent={screenShareComponent}
-        maxRemoteVideoStreams={maxRemoteVideoStreams}
-        dominantSpeakers={dominantSpeakers}
-        parentWidth={containerWidth}
-        styles={styles}
-      />
-    );
+  const layoutProps = useMemo(
+    () => ({
+      remoteParticipants,
+      /* @conditional-compile-remove(pinned-participants) */ pinnedParticipants,
+      screenShareComponent,
+      showCameraSwitcherInLocalPreview,
+      maxRemoteVideoStreams,
+      dominantSpeakers,
+      styles,
+      onRenderRemoteParticipant: onRenderRemoteVideoTile ?? defaultOnRenderVideoTile,
+      localVideoComponent: localVideoTile,
+      parentWidth: containerWidth,
+      parentHeight: containerHeight,
+      isLocalVideoFloating: layout === 'floatingLocalVideo'
+    }),
+    [
+      remoteParticipants,
+      screenShareComponent,
+      showCameraSwitcherInLocalPreview,
+      maxRemoteVideoStreams,
+      dominantSpeakers,
+      styles,
+      localVideoTile,
+      containerWidth,
+      containerHeight,
+      onRenderRemoteVideoTile,
+      defaultOnRenderVideoTile,
+      layout,
+      /* @conditional-compile-remove(pinned-participants) */ pinnedParticipants
+    ]
+  );
+
+  const videoGalleryLayout = useMemo(() => {
+    /* @conditional-compile-remove(pinned-participants) */
+    if (pinnedParticipants.length > 0) {
```

Comment:
You already have layoutProps as a dependency to this function.
Using `layoutProps.pinnedParticipants.length` allows you to remove the conditional dependency for `pinnedParticipants.length` ;)

Reply:
Ah nice. Save a dependency. Thanks.